isis/src/hayabusa/apps/amicacal/AmicaCalUtils.h

```diff
@@ -96,42 +97,54 @@ static void loadNaifTiming() {
  *
  * @return @b double Distance in AU between Sun and observed body.
  */
-static bool sunDistanceAU(const QString &scStartTime,
+static bool sunDistanceAU(Cube *iCube,
+                          const QString &scStartTime,
                           const QString &target,
                           double &sunDist) {
 
   //  Ensure NAIF kernels are loaded
-  loadNaifTiming();
-  sunDist = 1.0;
-
-  //  Determine if the target is a valid NAIF target
-  SpiceInt tcode;
-  SpiceBoolean found;
-  bodn2c_c(target.toLatin1().data(), &tcode, &found);
-
-  if (!found) return (false);
-
-  //  Convert starttime to et
-  double obsStartTime;
-  scs2e_c(-130, scStartTime.toLatin1().data(), &obsStartTime);
+  try {
+    Camera *cam; 
+    cam = iCube->camera();
+    cam->SetImage (0.5, 0.5);
+    iTime startTime(cam->time());
+    cam->setTime(startTime);
```

Comment:
Did you find this setTime to be necessary? If SetImage set this time then setTime of this time should not change anything.